### DISCUSSION TITLE
Deprecating SUAppcastitem is causing issue

### DISCUSSION THREAD

**Discussion by prasann-1433** [sparkle-project/Sparkle#2352] [Q&A]
### - (nullable instancetype)initWithDictionary:(NSDictionary *)dict 

in new version this api is been deprecated and this is causing an issue where we want user to updated to correct arm64 version of app who are using the emulated version in their M1 or M2 machine.  

In order to do that we use the handler 

bestValidUpdate(in appcast: SUAppcast, for updater: SPUUpdater) -> SUAppcastItem?

to update the SUAppcastItem and get the sparkle update the correct arm64 version. 

More over we use this to mock the delegate functions for our unit testing as well and they all are failing.  Appreciate if we can have them added back. 

### Sparkle's output from Console.app 
N/A 
```

```

### Steps to reproduce the behavior

[The more information provided and pasted verbatim, the easier it will be to diagnose an issue. If you can provide the affected application/binary and XML feed to reproduce an issue, share them]

**Comment by zorgiepoo** [comment]
> \- (nullable instancetype)initWithDictionary:(NSDictionary *)dict __deprecated_msg("Properties that depend on the system or application version are not supported when used with this initializer. The designated initializer is available in SUAppcastItem+Private.h. Please first explore other APIs or contact us to describe your use case.");

It was deprecated (note not removed) because creating appcast items this way does not support new appcast features or guarantee correct behavior (e.g. with minimum OS version requirements, which 1.x did not handle correctly).

We don't recommend creating appcast items or using `bestValidUpdate` when [channels](https://sparkle-project.org/documentation/publishing/#channels) (for beta versions) or [multiple feeds](https://sparkle-project.org/documentation/publishing/#setting-the-feed-programmatically) (for parallel trains, like arm64 vs x86_64, even though you should be using a universal binary instead) can work instead. And we more so do not recommend trying to create custom appcast items if at all possible.

If you want to migrate with minimal effort, there is a new initializer in `<Sparkle/SUAppcastItem+Private.h>`. It's Private and not our best supported path. Please see Upgrading to Sparkle 2.0 changes in https://sparkle-project.org/documentation/upgrading/ as well for bestValidUpdate changes.
